torchmultimodal/models/gpt.py

```diff
@@ -28,12 +29,278 @@ class TransformerLayerOutput(NamedTuple):
+        out_tokenizer (nn.Module): Tokenizer for the output modality. Must have methods ``encode``, ``decode``.
+        mm_decoder (nn.Module): Multimodal transformer decoder. An instace of
+            :py:class:`MultimodalTransformerDecoder`.
+        in_projection (nn.Module, optional): Projects the input modality token embeddings to match size of the
+            transformer decoder. Defaults to ``None``.
+        out_projection (nn.Module, optional): Projects the output modality token embeddings to match size of the
+            transformer decoder. Defaults to ``None``.
+        norm_layer (Callable[..., nn.Module], optional): Which normalization layer to use. Supports ``nn.Module`` or
+            partial. If ``None``, ``nn.LayerNorm`` will be used as the default.
+
+    Args:
+        in_tokens (Tensor, optional): Tensor of dimension ``(b, in_seq_len, c)`` containing tokens
+            for the input modality. Defaults to ``None``.
+        out_tokens (Tensor, optional): Tensor of dimension ``(b, out_seq_len, c')`` containing tokens
+            for the output modality. Defaults to ``None``.
+        in_pos_ids (Tensor, optional): Tensor of dimension ``(b, in_seq_len)`` containing indices for the
+            input modality position embeddings. Defaults to ``None``.
+        out_pos_ids (Tensor, optional): Tensor of dimension ``(b, out_seq_len)`` containing indices for the
+            output modality position embeddings. Defaults to ``None``.
+        attn_mask (Tensor, optional): Tensor of dimension ``(b, q_seq_len, k_seq_len)`` where prefixes ``q``
+            and ``k`` stand for query and key. Contains 1s for positions to attend to and 0s for masked positions.
+            Defaults to ``None``.
+        head_mask (Tensor, optional): Tensor of dimension ``(b, h, q_seq_len, k_seq_len)``. Masks need to be
+            specified for each attention head. Defaults to ``None``.
+        use_cache (bool, optional): If ``True``, caches past key/value tensors for faster decoding. If ``False``,
+            recomputes key and value for each decoding step. Defaults to ``False``.
+        causal (bool. optional): If ``True``, use causal attention. Defaults to ``False``.
+        right_shift (bool): If ``True``, shifts the embedding vectors to the right and prepends it with start of
+            sentence token. Defaults to ``False``. This option is disregarded during training mode
+        return_attn_weights (bool, optional): If ``True``, returns attention probabilities of each transformer
+            layer. Defaults to ``False``.
+        return_hidden_states (bool): If ``True``, returns the embeddings of each transformer layer. Defaults to
+            ``False``.
+
+    Raises:
+        AttributeError: If input tokenizer does not implement methods ``encode`` and ``lookup`` or if output
+            tokenizer does not implement methods ``encode``, ``lookup`` and ``decode``.
```

Comment:
One question here: do most tokenizers implement `lookup` method? Personally I haven't seen it before.

Reply:
Let me add a reference to VQVAE API since it has the lookup method.